### DISCUSSION TITLE
Anbindung Tuya Device

### DISCUSSION THREAD

**Discussion by DB1113** [fhempy/fhempy#139] [Q&A] [answered]
Hallo,

ich hab hier einen Eco-Dim.07 WIFI.
Den habe ich in der SmartLife App hinzugefügt. Der Dimmer lässt sich von der APP steuern.

Dann habe ich bei Tuya ein neues Cloud-Projekt erstellt und per Devices-->Link Tuya App Account versucht den SmartLife-Acc mit Tuya zu verbinden. Es wird dort (bei Tuya) kein Device und/oder verlinkter Account angezeigt. Ist das richtig?

Weiterhin findet dann tuya in fhem auch kein Device.

Könnte man mir bitte helfen herauszufinden, was ich falsch mache?
Vielen Dank!

Welche Infos werden benötigt?

**Comment by fhempy** [comment]
Was kommt für eine Meldung wenn du "Link Tuya App Account" machst? Nach dem Verlinken, muss der Account dort in der Liste erscheinen. Tut er das nicht, hat etwas nicht funktioniert.

**Comment by DB1113** [comment]
"Devices will be linked with the project
0 device(s) under the "智能生活" App Account being linked with this project. This process will be completed within 0 second(s)."

Wenn ich im selben Projekt den Acc nochmal hinzufügen will kommt das:
"The number of projects linked with this Tuya Smart App account has reached the upper limit of 2. Unlink the account from existing projects and try again."
Wie kann man die Projekte wieder in der App entfernen? Da wird es ja nirgends angezeigt... Und in Tuya auch nicht?!

**Comment by fhempy** [comment]
Bist du eh auf der richtigen Seite? Mach bitte mal einen Screenshot was du gemacht hast.

**Comment by DB1113** [comment]
![Tuya1](https://user-images.githubusercontent.com/107619866/226724293-f7d5e66d-77ba-4922-a269-e6f531b0e0c7.jpg)

Interessanter Weise steht da auch, dass kein Acc verlinkt ist. Aber da kommt nix...

**Comment by DB1113** [comment]
![Tuya2](https://user-images.githubusercontent.com/107619866/226725777-9fd65b48-b0c0-4c79-b84e-0a735519de70.jpg)

![Tuya3](https://user-images.githubusercontent.com/107619866/226726659-420f28c4-3de9-4f6b-a4af-759259e476b4.jpg)

![Screenshot_20230321-205402](https://user-images.githubusercontent.com/107619866/226726787-d5cd21eb-65d1-43d5-901d-21e4e54f2a62.jpg)

**Comment by fhempy** [comment]
In der SmartLife App hast du den QR Code gescannt, oder? ...und in der SmartLife App ist das Device in deinem Konto hinterlegt?

**Reply by DB1113** [reply]
Ja, 
in der App über Profil und dann oben rechts den Scanner auswählen.
Den Screenshot der App siehst du oben.

**Comment by fhempy** [comment]
Hm...dann machst du eigentlich alles richtig.

Schon mal probiert das Device in der App zu löschen und neu hinzuzufügen? Hast du noch ein anderes Device mit welchem du es probieren kannst?

**Reply by DB1113** [reply]
Habe schon den gesamten Acc in der App neu erstellt.. Neue E-Mail-Adresse usw. 
Bei Tuya im übrigen auch...
Nein, leider kein weiteres Device.

**Comment by DB1113** [comment]
Warum wird für die IP des Device eigentlich eine so komische IP angezeigt?
Nämlich 87.126.23.XXX
Mein Adressbereich ist völlig anders...?!

Also in der App meine ich jetzt.

**Reply by fhempy** [reply]
Das ist normal. Entweder ist es deine public IP oder die interne IP von tuya.

Mach bitte für deinen Fehler hier ein TIcket auf:
https://service.console.tuya.com/8/3/create?step=3&id=01030507&source=support_center

Im Normalfall antworten die innerhalb von ein paar Tagen und sind auch sehr lösungsorientiert.

**Reply by DB1113** [reply]
Hab ich gemacht...
Aber ob ich denen das mit meinem Englisch gut erklären kann...?! :|
Mal sehen.
Vielen Dank bis hier!

**Comment by DB1113** [comment]
Mit deiner Anleitung hier:
`Individual Device
If a device couldn't be setup properly or you know your local keys and don't need setup, you can define it on your own

define wifi_plug fhempy tuya PRODUCT_ID DEVICE_ID IP LOCAL_KEY [VERSION] [API_KEY] [API_SECRET]
PRODUCT_ID: Set to "api" if API_KEY and API_SECRET are provided, otherwise use the proper product id
DEVICE_ID: Take that one from the tuya_system device you created
IP: Check for the IP in your tuya_system device or on your router
LOCAL_KEY: Take that one from the tuya_system device you created
VERSION: Take that one from the tuya_system device you created (default 3.3)
API_KEY: If the product ID couldn't be found in the standard mappings, the possible functions will be retrieved from tuya cloud
API_SECRET: If the product ID couldn't be found in the standard mappings, the possible functions will be retrieved from tuya cloud`

funzt das wohl nicht?

Was wäre denn der LOCAL_KEY, bzw. wo kriegt man den her?
Wo findet man denn die PRODUCT_ID?
Die DEVICE_ID wird wohl die virtuelle ID des Devices aus der App sein, oder?

**Reply by fhempy** [reply]
Bekommt man alles über die Cloud.

**Comment by DB1113** [comment, marked as answer]
Hi,

den Server muss man ZWINGEND auf "Central Europe Data Center" bzw. auf den, welcher von der SmartLife-App verwendet wird, stellen!
Dann funktioniert es auch mit dem Nachbarn...

Danke für deine Unterstützung!

**Comment by wiresio** [comment]
Habe die Frage schon im [FHEM Forum](https://forum.fhem.de/index.php?topic=127441.885#886) gestellt:   
Wie funktioniert die "Offline" Erkennung? Wenn ich meine Tuya-Lampe per Wandschalter ausschalte, bleibt "online" auf 1 und "state" auf on.

Muss ich dazu ein Skript schreiben, dass die IP-Adresse pingt und über ein userReading "state" dementsprechend korrigiert? Oder ist so etwas schon im Binding drin?
